### DISCUSSION TITLE
voidtools Everything integration

### DISCUSSION THREAD

**Discussion by couleurm** [Hofknecht/SystemTrayMenu#375] [Q&A] [answered]
One great thing would be some way to integrate Everything's awesome instant system-wide search

**Comment by Hofknecht** [comment, marked as answer]
@couleurm 
Thanks for opening the discussion on this.
Of course, we also noticed the tool. It follows a slightly different idea than we do, but it also has a lot of similarities. We should look at the code and see if we can win something for ourselves there.

i think the github repo is this:
 https://github.com/stnkl/EverythingToolbar

a system-wide search initially shows too much for my taste. It would be very interesting to see how this is achieved, really finding all files on a system extremely quickly, I suspect some disadvantages.

**Comment by WouterVanGoey** [comment]
Have you looked into this? What I would find very useful is the simple functionality of starting a search in Everything from the SystemTrayMenu search bar. Maybe with Ctrl+Enter?

![image](https://user-images.githubusercontent.com/6655152/193553471-8728b114-2f34-494a-ac2f-ccf02ac3fe04.png)

**Comment by Hofknecht** [comment]
@WouterVanGoey 
Thank you for the feedback!

this functionality could be reached e.g. with an autohotkey script

```
Hotkey, ^!s, SearchViaEverything
return

SearchViaEverything:
Send,{Ctrl down}a{Ctrl up}
Sleep, 100
Send ^a^c ;control a then control c
clipboard = 
Send ^c
ClipWait  
Sleep, 100
Send, {LWin}
Sleep, 500
Send, Everything
Sleep, 100
Send, {enter}
Sleep, 500
send ^v ; control +v  (paste)
return
```

[search via everything by ctrl alt s.zip](https://github.com/Hofknecht/SystemTrayMenu/files/9697793/search.via.everything.by.ctrl.alt.s.zip)

(the sleep times could probably be optimized)
with that script you could use Ctrl+Alt+s to select searchtext via ctrl+a, copy it into clipboard, open everything via windows key and search everthing and press enter, then paste the searchtext via ctrl+v
would that solve your issue?

 or do you mean display the results of everything in the STM? (i assume that would be a long way)

**Reply by WouterVanGoey** [reply]
Well, that would be one way of doing it certainly, although I assume there is a way to interface with everything directly? I noticed you forked a project called Everything Toolbar and had a look at it. I'm using it now and it's pretty great and I will use it instead for now.  
I still think that some kind of merger of these two projects would be the ultimate search/utility tool.

**Reply by Hofknecht** [reply]
currently I don't know a way to interface directly with voidtools. we'll have to look again in the future, maybe we'll find a way to improve something

**Reply by topeterk** [reply]
The EverythingToolbar is using [native DLL calls to the Everything64.dll](https://github.com/stnkl/EverythingToolbar/blob/abba3705997f6da9a0b7380486943822abfad24b/EverythingToolbar/EverythingSearch.cs#L256) which seems to come from [voidtools.com](https://www.voidtools.com) (see the project's readme).
They seem to have CLI and SDK, so it *should* be possible to use their "search engine".
Haven't had a deeper look but it seems EverythingToolbar also stores completed searches in some sort of history XML to show cached results.

**Reply by topeterk** [reply]
PS: From my understanding they let the DLL load with e.g. explorer or something and then gather data in the background over time. By loading the DLL one gets access to the shared data and can access the database to "perform" the search by fetching the requested data from the cache.

**Reply by topeterk** [reply]
Thinking a bit more about this topic...
Why would it make sense, to do a system wide search integration within STM when there is EverythingToolbar already?
For a local search STM should be fine without further 3rd party search integration or is there any issue?
So from my current understanding it doesn't make sense spending time into an integration.
